Source/WebCore/rendering/RenderTheme.cpp

```diff
@@ -456,6 +457,140 @@ ControlPartType RenderTheme::autoAppearanceForElement(RenderStyle& style, const
     return ControlPartType::NoControl;
 }
 
+RefPtr<ControlPart> RenderTheme::createControlPartForRenderer(const RenderObject& renderer) const
+{
+    ControlPartType type = renderer.style().effectiveAppearance();
+
+    switch (type) {
+    case ControlPartType::NoControl:
+    case ControlPartType::Auto:
+        break;
+
+    case ControlPartType::Checkbox:
+    case ControlPartType::Radio:
+    case ControlPartType::PushButton:
+    case ControlPartType::SquareButton:
+    case ControlPartType::Button:
+    case ControlPartType::DefaultButton:
+    case ControlPartType::Listbox:
+    case ControlPartType::Menulist:
+    case ControlPartType::MenulistButton:
+        break;
+
+    case ControlPartType::Meter: {
+        const auto& renderMeter = downcast<RenderMeter>(renderer);
```

Comment:
`ASSERT(is<RenderMeter>(renderer))`?

This wasn't always the case before we implemented the new `appearance` spec, but I think it's safe now due to `RenderTheme::adjustAppearanceForElement`.

Reply:
The ASSERT was added.